python/lsst/ts/scheduler/driver/feature_scheduler.py

```diff
@@ -0,0 +1,443 @@
+from .survey_topology import SurveyTopology
+
+
+__all__ = ["FeatureScheduler", "NoSchedulerError", "NoNsideError"]
+
+
+class NoSchedulerError(Exception):
+    """Exception raised when scheduler is not defined in the feature scheduler
+    configuration.
+    """
+
+    pass
+
+
+class NoNsideError(Exception):
+    """Exception raised when nside is not defined in the feature scheduler
+    configuration.
+    """
+
+    pass
+
+
+class FeatureSchedulerTarget(DriverTarget):
+    """Feature based scheduler target.
+
+    Parameters
+    ----------
+    observation : `np.ndarray`
+        Observation produced by the feature based scheduler.
+
+    """
+
+    def __init__(self, observation, targetid=0):
+
+        self.observation = observation
+
+        super().__init__(
+            targetid=targetid,
+            band_filter=observation["filter"][0],
+            ra_rad=observation["RA"][0],
+            dec_rad=observation["dec"][0],
+            ang_rad=observation["rotSkyPos"][0],
+            num_exp=observation["nexp"][0],
+            exp_times=[
+                observation["exptime"][0] / observation["nexp"][0]
+                for i in range(observation["nexp"][0])
+            ],
+        )
+
+
+class FeatureSchedulerParameters(DriverParameters):
+    """Feature Scheduler driver parameters.
+    """
+
+    scheduler_config = pex_config.Field(
+        "Python script with the feature scheduler configuration. It must define "
+        "a `scheduler` object of the type `lsst.sims.featureScheduler.schedulers.Core_scheduler`.",
+        str,
+    )
+    force = pex_config.Field(
+        "Force reloading scheduler? If `False` and scheduler is already configured "
+        "it will not be overritten.",
+        bool,
+    )
+
+
+class FeatureScheduler(Driver):
+    """Feature scheduler driver.
+    """
+
+    def __init__(self, models, raw_telemetry, parameters=None):
+
+        self.scheduler = None
+
+        self.nside = None
+
+        self.conditions = None
+
+        self.index_gen = index_generator()
+
+        self.next_observation_mjd = None
+
+        self.almanac = Almanac()
+
+        super().__init__(models, raw_telemetry, parameters)
+
+    def configure_scheduler(self, config=None):
+        """This method is responsible for running the scheduler configuration
+        and returning the survey topology, which specifies the number, name
+        and type of projects running by the scheduler.
+
+        By default it will just return a test survey topology.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration, as described by ``schema/Scheduler.yaml``
+
+        Returns
+        -------
+        survey_topology: `lsst.ts.scheduler.kernel.SurveyTopology`
+
+        Raises
+        ------
+        RuntimeError:
+            If `config` does not have a `scheduler_config` attribute or if it
+            points to a non-existing file.
```

Comment:
In this method also NoSchedulerError and NoNsideError may be raised so please add those to the doc string.